src/jrd/pag.h

```diff
@@ -129,8 +129,13 @@ class PageSpace : public pool_alloc<type_PageSpace>
 	ULONG usedPages();
 	static ULONG usedPages(const Database* dbb);
 
+	struct ExtendResult
+	{
+		bool success = false;
+		ULONG pagesAllocated = 0;
+	};
 	// extend page space
-	bool extend(thread_db*, const ULONG, const bool);
+	ExtendResult extend(thread_db* tdbb, ULONG pageNum, bool forceSize);
```

Comment:
If it's possible to return `{success == true && pagesAllocated == 0}`, then I'd suggest to return `std::optional<ULONG>` instead of the new struct.